engine/lib/elgglib.php

```diff
@@ -1211,6 +1211,39 @@ function elgg_extract_class(array $array, $existing = [], $extract_key = 'class'
 	return array_values(array_unique($existing));
 }
 
+/**
+ * Calls a callable passing arguments and applying logic based on flags
+ *
+ * @param callable $callable  Callable to call
+ * @param array    $arguments Arguments to pass to callable
+ * @param null     $flags     Flags
+ *
+ * @return mixed
+ */
+function elgg_call(callable $callable, $arguments = null, $flags = null) {
```

Comment:
I'd say make $flags required and 1st argument, and no $argument. We really don't want devs to replace static analysis-friendly function calls with callbacks.